lib/Installation/ModuleRegistration/ModuleRegCodeController.pm

```diff
@@ -0,0 +1,60 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: The class introduces business actions for Module Registration Code dialog.
+#
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+package Installation::ModuleRegistration::ModuleRegCodeController;
+use strict;
+use warnings;
+use Installation::ModuleRegistration::ModuleRegCodePage;
+use Installation::Popups::UntrustedGPGPopup;
+use YuiRestClient;
+
+sub new {
+    my ($class, $args) = @_;
+    my $self = bless {}, $class;
+    return $self->init($args);
+}
+
+sub init {
+    my ($self, $args) = @_;
+    $self->{ModuleRegCodePage} = Installation::ModuleRegistration::ModuleRegCodePage->new({app => YuiRestClient::get_app()});
+    $self->{UntrustedGPGKeyPopup} = Installation::Popups::UntrustedGPGPopup->new({app => YuiRestClient::get_app()});
+    return $self;
+}
+
+sub get_module_regcode_page {
+    my ($self) = @_;
+    die "Module Registration Code page is not displayed" unless $self->{ModuleRegCodePage}->is_shown();
+    return $self->{ModuleRegCodePage};
+}
+
+sub get_untrusted_GPG_popup {
+    my ($self) = @_;
+    die "Untrusted GPG key popup is not displayed" unless $self->{UntrustedGPGKeyPopup}->is_shown();
+    return $self->{UntrustedGPGKeyPopup};
+}
+
+sub wait_regcode_page {
+    my ($self, $args) = @_;
+    YuiRestClient::Wait::wait_until(object => sub {
+            $self->{ModuleRegCodePage}->is_shown({timeout => 0});
+    }, %$args);
+}
+
+sub license_workstation_ext_module {
+    my ($self, $regcode) = @_;
+    $self->get_module_regcode_page()->set_we_code($regcode);
```

Comment:
we can wait explicitly about this line with `wait_regcode_page`. There was controversy about this, but I don't think we should reflect the waiting mechanism as part of the test, the disadvantage is that makes the method a bit more complex, but at least is clear that we pass it the timeout and we are doing some fancy waiting, not so usual, like the default one.